File: kotlin/simple-api/src/main/kotlin/mtakeshi1/githut/io/model/DataAccess.kt
package mtakeshi1.githut.io.model

interface DataAccess {

    fun listProducts(): List<Models.SmallProduct>
    fun productWithId(id: Int): Models.Product

    fun newOrder(items: List<Models.ShoppingCartEntry>): Models.ShoppingCart
    fun orderWithId(id: Int): Models.ShoppingCart

}

File: scala/scala-rest/app/src/main/scala/simplerest/DataAccess.scala
package simplerest

import cats.effect.IO

trait DataAccess {
  def listProducts(): IO[List[Models.ProductInformation]]
  def productWithId(id: Int): IO[Models.Product]
  def newOrder(items: List[Models.ShoppingCartEntry]): IO[Models.ShoppingCart]
  def orderWithId(id: Int): IO[Models.ShoppingCart]
}
